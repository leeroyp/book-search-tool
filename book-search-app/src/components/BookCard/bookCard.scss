// @mixin respond ($breakpoint) {

//     // $breakpoint is simply a variable that can have several values

//     @if $breakpoint==tablet {

//         // here `laptop` is the value of $breakpoint
//         // when call laptop, we mean the following piece of code        

//     @media only screen and (max-width: 600px) {
//       @content;
//     }
//   }

//   @if $breakpoint==mobile {
//     @media only screen and (max-width: 480px) {
//       @content;
//     }
//   }
// }


.container{
    width: 50vw;
    height: auto;
}

.Book-card{
    // width: 250px;
    height: 250px;
    display: flex;
    flex-direction: row;
    align-items: flex-start;

    &_cover{
        width: 185px;
        height: 250px; 
        margin-right:5vw;
    }

    &__title{
        font-family: sans-serif;
        font-weight: 800;
    }

    &__text{
        font-family: sans-serif;
        font-weight: normal; 
        font-size: 18px;
        line-height: 24px;  
    }
.contents{
display: flex;
flex-direction: column;
text-align: left;

}
}

