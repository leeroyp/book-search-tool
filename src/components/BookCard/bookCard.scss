@mixin respond ($breakpoint) {

  @if $breakpoint==desktop {
      @media only screen and (max-width: 1048px) {
        @content;
      }
    }

  @if $breakpoint==tablet {
  @media only screen and (max-width: 600px) {
    @content;
  }
}

@if $breakpoint==mobile {
  @media only screen and (max-width: 480px) {
    @content;
  }
}
}



.container{
    width: 50vw;
    height: auto;
}

.Book-card{
    width: 50vw;
    height: 250px;
    display: flex;
    flex-direction: row;
    align-items: flex-start;

    @include respond(tablet) {
      width: 75vw;
      height: auto;
   }

   @include respond(mobile) {
    width: 75vw;
    height: auto;
    }

    &_cover{
        width: 185px;
        height: 250px; 
        margin-right:5vw;

        @include respond(tablet) {
          width: 125px;
          height: 175px;
       }
    
       @include respond(mobile) {
        width: 125px;
        height: 175px;
        }
    }

    &__title{
        font-family: sans-serif;
        font-weight: 800;
    }

    &__text{
        font-family: sans-serif;
        font-weight: normal; 
        font-size: 18px;
        line-height: 24px;  
        @include respond(tablet) {
          font-size: 14px;
          line-height: 17px; 
       }
    
       @include respond(mobile) {
        font-size: 14px;
        line-height: 17px; 
        }

    }
.contents{
display: flex;
flex-direction: column;
text-align: left;

}
}

