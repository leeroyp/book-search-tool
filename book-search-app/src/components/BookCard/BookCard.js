import "./bookCard.scss";
import openlibrary from "../../images/openLibrary.jpeg";

const BookCard = ({ image, title, author, PubDate }) => {
  return (
    <div className="container">
      <div className="Book-card">
        <img
          src={
            image
              ? `http://covers.openlibrary.org/b/id/${image}-M.jpg`
              : openlibrary
          }
          alt={`${title} book cover`}
          className="Book-card_cover"
        />
        <div className="Book-card_contents">
          <h2 className="Book-card__title">{title}</h2>
          <p className="Book-card__text">By:{author}</p>
          <p className="Book-card__text">Year Published:{PubDate}</p>
        </div>
      </div>
      <hr />
    </ div>
  );
};

export default BookCard;
