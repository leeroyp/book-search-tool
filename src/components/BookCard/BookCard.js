import "./bookCard.scss";
import openlibrary from "../Images/openLibrary.jpeg";

const BookCard = ({ image, title, author, PubDate }) => {
  return (
    <>
      <div className="Book-card" data-testid="book-cards">
        <img
          data-testid="image"
          src={
            image
              ? `http://covers.openlibrary.org/b/id/${image}-M.jpg`
              : openlibrary
          }
          alt={`${title} book cover`}
          className="Book-card_cover"
        />
        <div className="Book-card_contents">
          <h2 className="Book-card__title">{title}</h2>
          <p>By: {author}</p>
          <p>Year Published: {PubDate}</p>
        </div>
      </div>
      <hr />
    </>
  );
};

export default BookCard;
