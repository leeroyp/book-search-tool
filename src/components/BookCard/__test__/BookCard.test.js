import React from "react";
import BookCard from "../BookCard";
import { render } from "@testing-library/react";

const exampleCard = {
  image: 10521270,
  title: "Harry Potter and the Philosopher's Stone",
  author: "J.K Rowling",
  PubDate: 1997,
};

it("matches snapshot", () => {
  const { asFragment } = render(
    <BookCard
      image={exampleCard.image}
      title={exampleCard.title}
      author={exampleCard.author}
      PubDate={exampleCard.PubDate}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
