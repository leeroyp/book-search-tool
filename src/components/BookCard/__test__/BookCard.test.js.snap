// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Book-card"
    data-testid="book-cards"
  >
    <img
      alt="Harry Potter and the Philosopher's Stone book cover"
      class="Book-card_cover"
      data-testid="image"
      src="http://covers.openlibrary.org/b/id/10521270-M.jpg"
    />
    <div
      class="Book-card_contents"
    >
      <h2
        class="Book-card__title"
      >
        Harry Potter and the Philosopher's Stone
      </h2>
      <p>
        J.K Rowling
      </p>
      <p>
        1997
      </p>
    </div>
  </div>
  <hr />
</DocumentFragment>
`;
